core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -377,5 +409,56 @@ public String location() {
     public void applyTo(TableMetadata.Builder metadataBuilder) {
       metadataBuilder.setLocation(location);
     }
+
+    @Override
+    public void applyTo(ViewMetadata.Builder viewMetadataBuilder) {
+      viewMetadataBuilder.setLocation(location);
+    }
+  }
+
+  class AddViewVersion implements MetadataUpdate {
+    private final ViewVersion viewVersion;
+
+    public AddViewVersion(ViewVersion viewVersion) {
+      this.viewVersion = viewVersion;
+    }
+
+    public ViewVersion viewVersion() {
+      return viewVersion;
+    }
+
+    @Override
+    public void applyTo(TableMetadata.Builder metadataBuilder) {
+      throw new UnsupportedOperationException(
+          String.format("Cannot apply update %s to a table", this.getClass().getSimpleName()));
+    }
+
+    @Override
+    public void applyTo(ViewMetadata.Builder viewMetadataBuilder) {
+      viewMetadataBuilder.addVersion(viewVersion);
+    }
+  }
+
+  class SetCurrentViewVersion implements MetadataUpdate {
+    private final int versionId;
+
+    public SetCurrentViewVersion(int versionId) {
+      this.versionId = versionId;
+    }
+
+    public int versionId() {
+      return versionId;
+    }
+
+    @Override
+    public void applyTo(TableMetadata.Builder metadataBuilder) {
+      throw new UnsupportedOperationException(
+          String.format("Cannot apply update %s to a table", this.getClass().getSimpleName()));
```

Comment:
An alternative is to make this the default implementation for `void applyTo(TableMetadata.Builder)` so that we don't need it in two places (or more if we add more view mutations).

Reply:
makes sense and I've done that